accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -673,6 +627,18 @@ The use of `typedef` for something which is not a type may be too confusing. Ano
 extension MyWidgetList<T extends Widget> = prefix.MyList<T>;
 ```
 
+### Avoiding Scope Conflicts
+
+We allow importing two extensions with the same name into the same scope, and both of them *work*. They cannot be referenced by name, but they can be implicitly invoked. This breaks the current invariant that we cannot have two declarations with the same name in the same scope.
+
+To retain this invariant, we can define a "fresh extension alias" for each extension declaration. It is a second *fresh name* for the extension (a name not occurring anywhere else in the program), which will be a fresh *private* name if the extension's name is private (or omitted, in which case we only need the one fresh name), and a fresh non-private name if the extension's name is non-private. Any implicit extension invocation uses the *fresh extension alias* name to refer to the extension (so an `expr.foo()` which resolves to a method on the extension  `Ext` is equivalent to `$fresh$Ext(expr).foo()`). 
```

Comment:
Why not just use a prefix, instead of inventing a new concept?  Import it once prefixed to get a name for it, and once unprefixed to bring the members into scope?

Reply:
The goal of the idea in this section is not to solve the reference problem, but to avoid having two elements with the same name in the same scope. I have rewritten other text so that the available extensions are not based on being "in scope" but rather on being among the members of an import declaration, whether that member is actually added to the library scope or not.

This was just a potential other approach, and it might not be necessary.

Reply:
I think prefixing solves this.  No matter what, you need to have a way of associating extension member names (of which there will be lots with the same name in "scope") with extension declarations.  All I'm saying is, don't do that association based on the unqualified name of the extension declaration: instead, do it based on a fully qualified name.  Then you never have to talk about there being two extension declarations with the same name in scope.  You just have things like `p0.E`, `p1.E` and `p1.E2` in scope, and if you see an unqualified reference to `E2` you resolve it to `p1.E2`, but if you see an unqualified reference to `E` you fail to resolve it.  And now you have whatever mechanism you use for resolving extension member names to extension declarations map `foo` to `p0.E.foo` and `bar` to `p1.E.bar`, and everybody is happy.